scripts/core/editor3/components/Editor3.jsx

```diff
@@ -4,32 +4,18 @@ import {connect} from 'react-redux';
 
 import {Editor3Component} from './Editor3Component';
 import {MultipleHighlights} from './MultipleHighlights';
-import {getHighlightsState, setHighlightsState, availableHighlights} from '../helpers/highlights';
 import * as actions from '../actions';
 
-function hadHighlightsChanged(prevEditorState, nextEditorState) {
-    return getHighlightsState(prevEditorState) !== getHighlightsState(nextEditorState);
-}
 
 export class Editor3Base extends React.Component {
     static getDecorator(disableSpellchecker) {
         return Editor3Component.getDecorator(disableSpellchecker);
     }
 
-    onHighlightChange(editorState, hightlightsState) {
-        const newEditorState = setHighlightsState(editorState, hightlightsState);
-
-        this.props.onChange(newEditorState);
-    }
-
     render() {
         var props = {
             ...this.props,
-            availableHighlights: availableHighlights,
-            onHighlightChange: this.onHighlightChange.bind(this),
-            hadHighlightsChanged: hadHighlightsChanged,
-            getHighlightsState: getHighlightsState,
-            setHighlightsState: setHighlightsState
+            onHighlightChange: this.props.onChange
```

Comment:
Might be better to call this simply `onChange` so we know it's a usual editor3 change handler and isn't really related to highlights.